src/main/java/frc/utils/time/TimeUtils.java

```diff
@@ -0,0 +1,50 @@
+package frc.utils.time;
+
+import edu.wpi.first.hal.HALUtil;
+import frc.utils.Conversions;
+import frc.utils.alerts.Alert;
+import frc.utils.alerts.AlertManager;
+import frc.utils.alerts.PeriodicAlert;
+import org.littletonrobotics.junction.Logger;
+
+public class TimeUtils {
+
+	public static final double DEFAULT_CYCLE_TIME_SECONDS = 0.02;
+
+	private static double lastCycleTimeSeconds = 0;
+	private static double newCycleTimeSeconds = 0;
+
+	static {
+		AlertManager.addAlert(
+			new PeriodicAlert(
+				Alert.AlertType.WARNING,
+				TimeConstants.LOG_PATH + "CycleOverrunAt",
+				() -> getCurrentCycleTimeSeconds() > DEFAULT_CYCLE_TIME_SECONDS + TimeConstants.TIME_STEP_TOLERANCE_SECONDS
+			)
+		);
+	}
+
+	public static void updateCycleTime(int roborioCycles) {
+		lastCycleTimeSeconds = newCycleTimeSeconds;
+		newCycleTimeSeconds = getCurrentTimeSeconds();
+
+		logStatus(roborioCycles);
+	}
+
+	private static void logStatus(int roborioCycles) {
+		double currentTimeSeconds = getCurrentTimeSeconds();
+
+		Logger.recordOutput(TimeConstants.LOG_PATH + "CycleTimeSeconds", getCurrentCycleTimeSeconds());
+		Logger.recordOutput(TimeConstants.LOG_PATH + "CurrentTimeSeconds", currentTimeSeconds);
+		Logger.recordOutput(TimeConstants.LOG_PATH + "AverageCycleTimeSeconds", currentTimeSeconds / roborioCycles);
+	}
+
+	public static double getCurrentTimeSeconds() {
+		return Conversions.microSecondsToSeconds(HALUtil.getFPGATime());
+	}
+
+	public static double getCurrentCycleTimeSeconds() {
```

Comment:
```suggestion
	public static double getLatestCycleTimeSeconds() {
```
just to make it more clear that this is the difference and not the current time

Reply:
סi dont agree its more clear, both names have same meanings

Reply:
I would like to hear more peoples opinions on this

Reply:
I agree with hillel, this indicates the difference between the previous cycle time and the current one, not something related to the current cycle happening.